virtual/activations.go

```diff
@@ -382,6 +384,47 @@ func (a *activations) getServerState() (
 	return a.serverState.serverID, a.serverState.serverVersion
 }
 
+func (a *activations) close(ctx context.Context, numWorkers int) error {
+	log.Print("acquiring lock for closing actor activations")
+	a.Lock()
+	log.Print("acquired lock for closing actor activations")
+	defer a.Unlock()
+
+	sem := semaphore.NewWeighted(int64(numWorkers))
+
+	wg := sync.WaitGroup{}
+	closed := int64(0)
+	expected := int64(len(a._actors))
+	for actorId, futActor := range a._actors {
+		sem.Acquire(ctx, 1)
+		wg.Add(1)
+		go func(actorId types.NamespacedActorID, futActor futures.Future[*activatedActor]) {
+			defer sem.Release(1)
+			defer wg.Done()
+
+			actor, err := futActor.Wait()
+			if err != nil {
+				log.Printf("failed to wait for an actor: %s", err.Error())
+				return
+			}
+
+			if err := actor.close(ctx); err != nil {
+				log.Printf("failed to close an actor: %s", err.Error())
```

Comment:
"failed to close actor during activations clean shutdown" + add actor ID in log message (maybe wait until rebasing on slog?)